paddle/fluid/framework/data_layout_transform.cc

```diff
@@ -185,11 +189,17 @@ void innerTransDataLayoutFromMKLDNN(DataLayout in_layout, DataLayout out_layout,
   } else {
     out->ShareDataWith(in);
   }
+  // For exepected NHWC data format we need to reshape the Output tensor
+  // As MKL-DNN description was in NCHW and paddle is expecting NHWC
+  if (out_layout == DataLayout::kNHWC) {
+    std::rotate(out_tz.begin() + 1, out_tz.begin() + 2, out_tz.end());
+    out->Resize(framework::make_ddim(out_tz));
+  }
```

Comment:
can we put this dim description transfer together? I mean I saw similar change in data_transform.cc? can we put together?

Reply:
Here we rotate dims *left* and in data_transform we rotate dims *right* so code is a bit different. As they are not exactly the same If we make function doing such a rotation it is likely to have two separate implementations rather than one unified.

Reply:
yes, I know it is different, I mean can we move these kinds of code together, in one file or better in one function. e.g in data_transform.cc::TransformData